### PR TITLE
L10n:de: help_ch_Tools_Assistant: The "Loan Repayment Calculator"

### PR DESCRIPTION
**Pull request by CWehli**
Loan Repayment Calculator

[Edit]: The further processing of this section will take place after the decision regarding the further development of the dialogue.

### CODE REVIEW

File: manual/de/Help_ch_Tools_Assistants.xml [hunk 1]
```diff
@@ -5307,101 +5307,175 @@
...
+            Feld die Eingabe zurückgesetzt haben. Ansonsten gibt es nichts zu berechnen.
+          </para>
+        </glossdef>
+      </glossentry>
+    </glosslist>
 
-    <para>The right pane contains buttons to select what sort of payments and compounding is used for the left
-      pane calculations.
+    <para>Im rechten Bereich sind Steuerelemente enthalten, mit denen Sie angeben können, welche Vorgaben
+      für die Aufzinsung und den Zahlungszeitpunkt für die Berechnungen im linken Bereich gelten.
```
Comment (fellen) [inline manual/de/Help_ch_Tools_Assistants.xml:5406]: Ich wäre eher für Verzinsung. Es gibt ja auch abgezinste Anleihen, die sich im Prinzip auch damit berechen lassen sollten.

**Review by fellen — COMMENTED** [review]
Habe mal grob  drübergeguckt.
Vielleicht noch mal aus der Benutzer-Perspektive überdenken:
- wofür ist das gut/was kann ich alles damit machen?
- Wie kann ich Ziel A, B,…erreichen?

File: manual/de/Help_ch_Tools_Assistants.xml [hunk 2]
```diff
@@ -5307,101 +5307,175 @@
...
+        <mediaobject>
+          <imageobject>
+            <imagedata fileref="figures/Tools_LoanCalculator.png" srccredit="Christian Wehling" />
+          </imageobject>
+        </mediaobject>
+      </screenshot>
+    </figure>
+
+    <para>Der Rechner ist in zwei Bereiche unterteilt, der eine dient zur <guilabel>Berechnung</guilabel> der
+      Darlehenswerte und der andere der Angabe von <guilabel>Zahlungsraten Optionen</guilabel>.
```
[the remark this answers is not in the document]
Comment (CWehli) [reply manual/de/Help_ch_Tools_Assistants.xml:5333]: > `<guilabel>Zahlungsraten Optionen</guilabel>` sollten wir im Programm ändern: Mindestens mit Bindestrich, aber Zahlungsraten ist auch ungwöhnlich.

Vielleicht passt als Überschrift `Zahlungsoptionen` oder `Zinsmodelle` besser? Gib mal einen kurzen Hinweis, dann nehme ich in Weblate die Korrektur vor.

File: manual/de/Help_ch_Tools_Assistants.xml [hunk 3]
```diff
@@ -5307,101 +5307,175 @@
...
 
-    <itemizedlist>
-      <listitem>
-        <para><emphasis role="bold">Payment Periods:</emphasis> This field is used to select the number of
-          payments you wish to use in the calculation.
-        </para>
-      </listitem>
+    <para>Der linke Bereich enthält fünf Eingabefelder mit jeweils einem
+      <guibutton>Zurücksetzen</guibutton> Knopf um die Eingabe in dem jeweiligen Feld zu löschen.
+    </para>
```
[the remark this answers is not in the document]
Comment (CWehli) [reply manual/de/Help_ch_Tools_Assistants.xml:5338]: Den `glossterm` habe ich mal eingetragen, auf `guibutton` habe ich verzichtet. Jetzt müssten wir allerdings noch irgendwo eine `glossdef` formulieren und einen Link setzen.

File: manual/de/Help_ch_Tools_Assistants.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (fellen) [reply manual/de/Help_ch_Tools_Assistants.xml]: @jralls From my POV `<b>Payment Options</b>` is wrong as `<b>Compounding</b>` is no payment.
Perhaps `Loan Options` fits better?

Off topic: Markups should be converted to glade attibutes, ideally with a patch of all po files.

@CWehli Darlehensoptionen scheint mir derzeit angemessen.

File: manual/de/ch_Tools_Assistants.xml [hunk 4]
```diff
@@ -6089,101 +6089,186 @@ tips_appendix.xml and to adapt the text here. -->
...
+                <term>Typ</term>
 
-      <listitem>
-        <para><guibutton>Continuous Compounding:</guibutton> This button is used when the interest is charged
-          continuously.
-        </para>
-      </listitem>
+                <listitem>
+                  <para>Wählen Sie als Zinstyp eine <guilabel>Schrittweise</guilabel> oder eine
+                    <guilabel>Kontinuierliche</guilabel> Verzinsung.
```
Comment (fellen) [inline manual/de/ch_Tools_Assistants.xml:6214]: Und was bedeuten die?

File: manual/de/ch_Tools_Assistants.xml [hunk 5]
```diff
@@ -6395,11 +6480,11 @@ tips_appendix.xml and to adapt the text here. -->
         <title>Lots in Account window for Security Capital Gains</title><screenshot id="tool-lotsInAcctStk">
           <mediaobject>
             <imageobject role="html">
-              <imagedata fileref="figures/Tool_LotsStk.png"
+              <imagedata fileref="figures/Help_Tool_LotsStk.png"
```
Comment (fellen) [inline manual/de/ch_Tools_Assistants.xml:6483]: Hast du das nicht rebased? Im folgenden auch.
Ist Geerts Bildercheck doch nicht so schlecht.

**Review by fellen — COMMENTED** [review]
Ich würde vorschlagen, zunächst den Dialog gemäß https://github.com/Gnucash/gnucash-docs/pull/261#discussion_r882431008 auf Vordermann zu bringen und dann die Acceleratoren etc. gleich hier einzupflegen.

File: manual/de/ch_Tools_Assistants.xml [hunk 6]
```diff
@@ -6089,101 +6089,186 @@ tips_appendix.xml and to adapt the text here. -->
...
-    <title>Loan Repayment Calculator</title>
+    <title>Darlehensrechner</title>
 
-    <para>The <guilabel>Loan Repayment Calculator</guilabel> is used to calculate compound interest. It
-      provides a way of entering four of the five parameters of a compound interest calculation and
-      then calculating the remaining figure.
+    <abstract>
+      <para>Der <guilabel>Darlehensrechner</guilabel> dient der Berechnung eines Rückzahlungsplans. Er bietet die
+        Möglichkeit, vier von fünf Parameter einer Zinseszinsberechnung vorzugeben und dann den
+        verbleibenden Parameter zu berechnen.
```
Comment (fellen) [inline manual/de/ch_Tools_Assistants.xml:6097]: …dient der **unverbindlichen** Berechnung…

Damit ist er eine gute Vorbereitung für den Link(Hypotheken-Rechner), der aus diesen und weiteren Angaben geplante Buchungen erstellt.

**Comment by jralls** [comment]
> from my POV <b>Payment Options</b> is wrong as <b>Compounding</b> is no payment.
Perhaps Loan Options fits better?

@fellen, I don't understand what you're on about. "Payment Options" does not appear in https://github.com/Gnucash/gnucash-docs/blob/maint/manual/C/ch_Tools_Assistants.xml.

> Off topic: Markups should be converted to glade attibutes, ideally with a patch of all po files.

*Way* off topic, since we don't use either pango or glade in the docs. Regardless, I disagree: Glade is a convenience tool, not a dependency. If it's introducing glade-specific noise into the builder files then that's a bug in Glade. It's bad enough that it sticks its version info into the builder files, but at least GtkBuilder and regular XML editors can ignore that.

**Comment by fellen** [comment]
> > from my POV Payment Options is wrong as Compounding is no payment.
> > Perhaps Loan Options fits better?
> 
> @fellen, I don't understand what you're on about. "Payment Options" does not appear in https://github.com/Gnucash/gnucash-docs/blob/maint/manual/C/ch_Tools_Assistants.xml.

In https://www.gnucash.org/docs/v4/C/gnucash-help/tool-calc.html is currently only <q>The right pane contains buttons to select what sort of payments and compounding is used for the left pane calculations.</q> Christian tried to describe the dialog with more details and I stumbled over <q>Zahlungsraten Optionen</q>.

This right column has the label [&lt;b&gt;Payment Options&lt;/b&gt;](https://github.com/Gnucash/gnucash/blob/d3a3e3e9f9a8db3294fb115522993f409fa3b44f/gnucash/gtkbuilder/dialog-fincalc.glade#L514). The replacement of the markup is described in 2. of https://wiki.gnucash.org/wiki/GUI_Guidelines#Markups_in_Text and no problem. 

But if I look at de.po, I see
```
#: gnucash/gtkbuilder/dialog-fincalc.glade:514
msgid "<b>Payment Options</b>"
msgstr "<b>Zahlungsraten Optionen</b>"
```
In a first approach most of us would translate it with <q>Zahlungsoptionen</q> and only  after seeing it in the dialog try to improve it.
Also the name of the glade file is not very well choosen and gives no clue.

File: manual/de/ch_Tools_Assistants.xml [hunk 7]
```diff
@@ -6089,101 +6089,186 @@ tips_appendix.xml and to adapt the text here. -->
...
+
+      <para>Gestartet wird der Rechner im Menü unter &mc.to.loan;.
+      </para>
+    </formalpara>
+
+    <figure pgwide="1">
+      <title>Der Darlehensrechner</title><screenshot>
+        <mediaobject>
+          <imageobject>
+            <imagedata fileref="figures/Tools_LoanCalculator.png" srccredit="Christian Wehling" />
```
[the remark this answers is not in the document]
Comment (CWehli) [reply manual/de/ch_Tools_Assistants.xml:6112]: @fellen, gute Frage, eine ältere Version habe ich noch gefunden. Den Dialog muss ich ja auch noch überarbeiten.

**Comment by jralls** [comment]
> In a first approach most of us would translate it with "Zahlungsoptionen" and only after seeing it in the dialog try to improve it.

Which is why I think that all short-phrase msgids should have context.  You don't like that because it greatly increases the number of msgids, to which I counter that in some languages those different msgids will have different translations. We already know of at least one example in Mandarin Chinese.

That said I agree that compounding is a characteristic of the loan, and that the frequency and whether payments are made at the beginning or end of the period are terms, not options. I'd like to see the dialog redesigned with Present Value and Future Value at the top followed by the rate, compounding radio, and compounding frequency listbox, followed by a block with periods, periodic payment, the when paid radio, and the calendar frequency of the payments. Precision goes at the bottom with the calculate button and the gross payout. 

The controls also need tooltips and the calculate button needs to be connected to the controls' changed signals: It currently lights up only if the user clicks one of the clear buttons. Better yet, get rid of it and the clear buttons and just hook the calculate function up to the changed signals.

> Also the name of the glade file is not very well choosen and gives no clue.

On the contrary: The glade file name is correct because it corresponds to the C files that define the control, gnucash/gnome/dialog-fincalc.[ch]. It's the title that's poorly chosen. While it's not as versatile as my trusty [HP-12C](https://en.wikipedia.org/wiki/HP-12C) it can already solve for any of the terms so it can be used for more than just loan repayments.

**Comment by fellen** [comment]
@Bob-IT perhaps a task for you?

> I'd like to see the dialog redesigned with Present Value and Future Value at the top followed by the rate, compounding radio, and compounding frequency listbox, followed by a block with periods, periodic payment, the when paid radio, and the calendar frequency of the payments. Precision goes at the bottom with the calculate button and the gross payout.
> 
> The controls also need tooltips and the calculate button needs to be connected to the controls' changed signals: It currently lights up only if the user clicks one of the clear buttons. Better yet, get rid of it and the clear buttons and just hook the calculate function up to the changed signals.

**Comment by fellen** [comment]
> Which is why I think that all short-phrase msgids should have context. You don't like that because it greatly increases the number of msgids, to which I counter that in some languages those different msgids will have different translations.

In theory you are right, but as long as we use only one meaning, a translator comment should be appropriate. But in both cases the coder or someone else later needs to see the ambiguity.

> We already know of at least one example in Mandarin Chinese.

Which one?

File: manual/de/Help_ch_Tools_Assistants.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (CWehli) [reply manual/de/Help_ch_Tools_Assistants.xml]: @fellen ,
> @CWehli Darlehensoptionen scheint mir derzeit angemessen.

Was hältst du von "Konditionen" oder "Darlehenskonditionen" ?

[the remark this answers is not in the document]
Comment (fellen) [reply manual/de/Help_ch_Tools_Assistants.xml]: Mit "Konditionen" könnte ich leben.
Der ganze Dialog läuft aber mehr auf [Finanzrechner](https://en.wikipedia.org/wiki/Financial_calculator) hinaus, nicht nur Darlehen.

**Pull request closed without merging**